erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -45,6 +48,8 @@ def validate(self):
 		self.validate_drop_ship()
 		self.validate_serial_no_based_delivery()
 		validate_inter_company_party(self.doctype, self.customer, self.company, self.inter_company_order_reference)
+		self.validate_batch_item()
+		validate_delivery_window(self, "validate")
```

Comment:
Hmm, I'm wondering if we can put this in the selling controller, and call it directly instead

Reply:
lets have a call about this